c_glib/arrow-glib/array-builder.cpp

```diff
@@ -3328,13 +3328,36 @@ gboolean
 garrow_string_array_builder_append_string(GArrowStringArrayBuilder *builder,
                                           const gchar *value,
                                           GError **error)
+{
+  return garrow_string_array_builder_append_string_n(builder,
+                                                     value,
+                                                     static_cast<gint32>(strlen(value)),
+                                                     error);
+}
+
+/**
+ * garrow_string_array_builder_append_string_n:
```

Comment:
Could you use `_len` suffix instead of `_n` suffix because GLib uses `_len` suffix for this case such as `g_string_append_len()` https://docs.gtk.org/glib/method.String.append_len.html ?

```suggestion
 * garrow_string_array_builder_append_string_len:
```